logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/DeleteMessageUseCase.kt

```diff
@@ -0,0 +1,57 @@
+package com.wire.kalium.logic.feature.message
+
+import com.benasher44.uuid.uuid4
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.client.ClientRepository
+import com.wire.kalium.logic.data.conversation.ConversationId
+import com.wire.kalium.logic.data.message.Message
+import com.wire.kalium.logic.data.message.MessageContent
+import com.wire.kalium.logic.data.message.MessageRepository
+import com.wire.kalium.logic.data.user.UserRepository
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.functional.suspending
+import com.wire.kalium.logic.sync.SyncManager
+import kotlinx.coroutines.flow.first
+import kotlinx.datetime.Clock
+
+class DeleteMessageUseCase(
+    private val messageRepository: MessageRepository,
+    private val userRepository: UserRepository,
+    private val clientRepository: ClientRepository,
+    private val syncManager: SyncManager,
+    private val messageSender: MessageSender
+) {
+
+    suspend operator fun invoke(conversationId: ConversationId, messageId: String, deleteForEveryOne: Boolean): Either<CoreFailure, Unit> {
```

Comment:
Can we have a typealias for `messageId`?

Reply:
I'm afraid not getting your point!

Reply:
I was thinking we could introduce a typealias for the `messageId` in the same way we do for the clientID to get some better type checking.
```
typealias ClientId = PlainId
```

But I see now that we deal with strings everywhere so I'm fine with doing tackling this in another PR.